web/pages/onboarding.tsx

```diff
@@ -49,13 +244,57 @@ export default function Onboarding() {
     }
   }
 
+  let answersState = [];
+  for (let i = 0; i < questionData.length; i++) {
+    if (questionData[i].dropdown) {
+      let ddAns = Array(questionData[i].options.length).fill(false);
+      answersState.push(ddAns);
+    } else {
+      answersState.push(Array(questionData[i].options.length).fill(false));
+    }
+  }
+
+  const [answers, setAnswers] = useState(answersState);
```

Comment:
Consider storing answers in an object with keys that match something like 

```js
{
  type: FosterType[];
  size: Size[];
  gender: Gender[];
  age: Age[];
  temperament: Temperament[];
  goodWith: GoodWith[];
  medical: Medical[];
  behavioral: Behavioral[];
  houseTrained: Trained[];
  crateTrained: Trained[];
  spayNeuterStatus: Status[];
}
```
^ these are stripped from `IPost` and I haven't cross-checked if the types are appropriate for each field so don't rely too much on my on-a-whim judgment

This will allow us to write to the database with more ease since they'll match the schema better when we eventually have the set up for the User document.

Reply:
To make this work with the code you currently have, you could consider making another key in your JS object that determines how it's stored in state, see below:

```js
{
      key: "temperament", // <-- this line
      title: "What temperament can you foster?",
      description: "",
      options: ["Friendly", "Scared", "Active", "Calm"],
      qtype: QType.Question,
      singleAnswer: false,
      dropdown: false,
      tooltip: "",
    },
```

would have an output object of
```js
{
      temperament: Temperament[],
      // ...
}
```